crypto/crypto.go

```diff
@@ -20,8 +22,38 @@ func GetPubKeyHex(privateKey PrivateKey) string {
 	return hex.EncodeToString(privKey.PublicKey.X) + hex.EncodeToString(privKey.PublicKey.Y)
 }
 
-func GeneratePrivateKey() (PrivateKey, error) {
-	return elliptic.GeneratePrivateKey(elliptic.Secp256k1)
+func GeneratePrivateKey(difficulty int) PrivateKey {
+	for true {
+		privKey, err := elliptic.GeneratePrivateKey(elliptic.Secp256k1)
+		if err != nil {
+			fmt.Println("Error while generating private key", err)
+			return nil
+		}
+
+		xPart := make([]byte, len(privKey.PublicKey.X))
+		copy(xPart, privKey.PublicKey.X)
+		pubKey := sha256.Sum256(append(xPart, privKey.PublicKey.Y...))
+
+		nullBytes := difficulty / 8
```

Comment:
Make this whole difficulty check in a separate function and write a test for it.